### PR TITLE
[DB TimeLock, 2021] Part 1: Another Look At TimeLock Migration ETE Tests

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- For various internal reasons, we expect to have another fairly large swathe of migrations from embedded timelocks to external timelock services. If we're going to do this, I would prefer for us to have working end-to-end tests.

**Implementation Description (bullets)**:
This PR can be viewed as consisting of a few components:
1. Refactoring of TimeLockMigrationEteTest to support more general test contexts.
2. The writing of a test context for DbKVS -> Db-TimeLock.
3. Changes to our Circle configuration to run this in a separate container because the tests are heavyweight.

**Testing (What was existing testing like?  What have you done to improve it?)**:
This PR focuses on tests.

**Concerns (what feedback would you like?)**:
- Spinning up another container for parallelism costs $$$, but I think it's worthwhile.
- I haven't written a fully parameterized test to run it for both because that would make builds much longer unless we did even more janky circle optimisation.

**Where should we start reviewing?**:
Start with TimeLockMigrationEteTest.

**Priority (whenever / two weeks / yesterday)**: this week please

### CODE REVIEW

File: scripts/circle-ci/run-circle-tests.sh [hunk 1]
```diff
@@ -55,7 +57,7 @@ JAVA_GC_LOGGING_OPTIONS="${JAVA_GC_LOGGING_OPTIONS} -XX:-TraceClassUnloading"
 JAVA_GC_LOGGING_OPTIONS="${JAVA_GC_LOGGING_OPTIONS} -Xloggc:build-%t-%p.gc.log"
 
 # External builds have a 16gb limit.
-if [ "$CIRCLE_NODE_INDEX" -eq "7" ]; then
+if [ "$CIRCLE_NODE_INDEX" -eq "9" ]; then
```
Comment (gmaretic) [inline scripts/circle-ci/run-circle-tests.sh:60]: Why did we change this from 7 to 9?

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/startup/TimeLockMigrator.java [hunk 2]
```diff
@@ -71,9 +76,14 @@ protected synchronized void tryInitialize() {
         try {
             destination.ping();
         } catch (Exception e) {
-            throw new AtlasDbDependencyException("Could not contact the Timelock Server.", e);
+            throw new AtlasDbDependencyException("Could not contact the TimeLock Server.", e);
         }
         long currentTimestamp = source.backupAndInvalidate();
+        log.info(
+                "Now fast forwarding the timestamp on TimeLock Server, possibly as part of a TimeLock migration."
+                        + " If you are already using TimeLock, DON'T PANIC if this number is less than what you"
```
Comment (gmaretic) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/factory/startup/TimeLockMigrator.java:84]: nit : DO NOT

File: atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/TimeLockMigrationEteTest.java [hunk 3]
```diff
@@ -42,12 +51,55 @@
...
+                        .dockerComposeYmlFile(new File("docker-compose.timelock-migration.cassandra.yml"))
+                        .databaseServiceName("cassandra")
+                        .build();
+                TEST_CONTEXT = ImmutableTimeLockMigrationTestContext.builder()
+                        .eteConfigWithEmbedded(embeddedConfig)
+                        .eteConfigWithTimeLock(new File("docker/conf/atlasdb-ete.timelock.cassandra.yml"))
+                        .dockerClientConfiguration(dockerClientConfiguration)
+                        .build();
+                break;
+            case POSTGRES_DB_TIMELOCK:
```
Comment (gmaretic) [inline atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/TimeLockMigrationEteTest.java:74]: So this is basically a placeholder for "we will parameterize on this once we want to run the test on posgres as well"? This whole static block is a bit of a mess, you could have just put it all as separate methods of the enum though not sure if you agree that's cleaner :P

**Review by gmaretic — APPROVED** [review]

File: scripts/circle-ci/run-circle-tests.sh (hunk 1 above)
Comment (jeremyk-91) [reply scripts/circle-ci/run-circle-tests.sh:60]: discussed offline: this is the errorprone container, and when we added container 8 we forgot to update this.

File: atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/TimeLockMigrationEteTest.java (hunk 3 above)
Comment (jeremyk-91) [reply atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/TimeLockMigrationEteTest.java:74]: There's an issue with shutting down the cluster on Postgres being unstable. This only applies to the shutdown part (not the assertions), and I don't have the bandwidth to fix that right now.

**Pull request merged**